0013-delete-enqueued-updates.md

```diff
@@ -0,0 +1,151 @@
+
+
+# Delete enqueued updates
+
+## 1. Feature Description and Interaction
+
+### Summary
+
+Allow to delete messages in the update queue or to empty the entire queue. A deleted update will have the status `aborted`.
+
+Questions not answered:
+- Should we delete in-process updates? If yes, what the default behavior for the clear method. What are the side effects?
+- We have an attribute on update info `processedAt`. What becomes this attribute if we add the possibility to abort an update? We already use this term in the case of a failed update.
+
+### Motivation
+
+The update queue is convenient because it allows us to serialize updates and assure the [ACID property](https://en.wikipedia.org/wiki/ACID) of MeiliSearch. It also causes some problems. As the queue message is an append-only queue, it is impossible to remove an update from this queue. The updates can be large and can take a long time to index. Sometimes, it would be convenient to delete a message in this queue or empty it. 
+
+### Additional Materials
+
+As explained on the [parent spec](https://github.com/meilisearch/specifications/pull/12), It's quite rare to have this kind of update queue. Elasticsearch or Algolia, who have this king of update queue, don't give the delete or clear their 'tasks' queues. 
+
+### Explanation
+
+To make this feature possible, two new routes and a new status (`aborted`) must be added. 
+
+An aborted update will stay at its place on the queue. So the typical answer of the get all update status route will be: 
+
+```json
+[
+    {
+        "status": "processed",
+        "updateId": 1,
+        ...
+    },
+    {
+        "status": "failed",
+        "updateId": 2,
+        ...
+    },
+    {
+        "status": "processed",
+        "updateId": 3,
+        ...
+    },
+    {
+        "status": "enqueued",
+        "updateId": 4,
+        ...
+    },
+    {
+        "status": "aborted",
+        "updateId": 5,
+        ...
+    },
+    {
+        "status": "enqueued",
+        "updateId": 6,
+        ...
+    },
+    ...
+]
+```
+
+the update status document will look like:
+
+```json
+{
+  "status": "aborted",
+  "updateId": 12,
+  "type": {
+    "name": "DocumentsAddition",
+    "number": 4
+  },
+  "duration": 0,
+  "enqueuedAt": "2019-12-07T21:16:09.623944Z",
+  "processedAt": "2019-12-07T21:16:09.703509Z" // Date of deletion
+}
+```
+
+
+
+#### HTTP API
+
+**Delete an update:**
+
+- Method: DELETE
+- Route: `/indexes/:index_uid/updates/:update_id`
+    - index_uid: The index uid
+    - update_id: The update identifier
+- Status Code: 204
+- Response: No Body
+
+Other proposition:
+
+- Status Code: 200
+- Response: 
+```json
+{
+    deletedUpdate: 12
+}
+```
+
+**Clear all update:**
+
+- Method: DELETE
+- Route: `/indexes/:index_uid/updates`
+    - index_uid: The index uid
+- Status Code: 200
+- Response: 
+
+```json
+{
+    nbUpdatesDeleted: 12
+}
+```
+
+or
+
+```json
+{
+    deletedUpdates: [12, 13, 14, 15]
+}
+```
```

Comment:
Same here as my comment above: I would rather the `204`

Reply:
I'm asking myself about the fact the user could need to have the list of deleted update or the number of deleted updates. Do you think it's not useful and too much? Should we choose to return 204 in every delete/clear request? Perhaps :D

Reply:
Following this StackOverflow response, there is no hard standardization for the `DELETE` route of a RESTful API: https://stackoverflow.com/questions/25970523/restful-what-should-a-delete-response-body-contain/25970628
So we can do what we want as long as it is consistent with our other API routes.

However, IMO it's too much for a MVP, not sure this is really useful to know the list (or the total number) of the deleted objects. It could have been useful maybe for the documents, but `documents` routes are asynchronous tasks, so this is not the same context.
For the updates, I think this is too much for a first introduction, especially because it would involve changing the `DELETE indexes/` route to make it consistent, for a user need we are not even sure. Maybe should we wait for the user requests to know their use case? This way we could adapt MeiliSearch in the best possible way.